src/cloudWatchLogs/commands/addLogEvents.ts

```diff
@@ -38,13 +39,18 @@ export async function addLogEvents(
             getLogger().debug('Update done, releasing lock...')
         })
     } catch (e) {
-        // contingency in case lock isn't busy but still locked out. Don't want to accidentally trigger making codelens not busy
-        getLogger().debug(`addLogEvents already locked for lock: ${lockName}`)
-        return
-    }
-
-    registry.setBusyStatus(uri, false)
-    if (onDidChangeCodeLensEvent) {
-        onDidChangeCodeLensEvent.fire()
+        if (CancellationError.isUserCancelled(e)) {
+            getLogger().debug('cwl: User Cancelled Search')
+        } else {
+            // TODO: We assume here that non-cancellation errors are always busty errors. This may not be true.
+            // contingency in case lock isn't busy but still locked out. Don't want to accidentally trigger making codelens not busy
+            getLogger().debug(`addLogEvents already locked for lock: ${lockName}`)
+            return
```

Comment:
We should show an error message to maintain the original behavior, something like this maybe:
```suggestion
            const error = e as Error
            vscode.window.showErrorMessage(
                localize(
                    'AWS.cwl.searchLogGroup.errorRetrievingLogs2',
                    'Error retrieving logs for {0}: {1}',
                    uri.path,
                    error.message
                )
            )
```

Reply:
I thought the original behavior was simply the early return so that it essentially ignored errors here. What about the busy/locked error? Do I want to display that to the user?

Reply:
> I thought the original behavior was simply the early return so that it essentially ignored errors here

`updateLog` used to catch everything, now it doesn't. That's why the original catch block is so odd.

> What about the busy/locked error?

The comment about this seems really vague. If this could even happen then something is wrong with the implementation.